tools/ci-build/runtime-versioner/src/command/patch.rs

```diff
@@ -0,0 +1,187 @@
+        smithy_rs
+            .cmd(
+                "./gradlew",
+                // limit services down to minimum required to reduce generation time
+                ["-Paws.services=+sts,+sso,+ssooidc", "aws:sdk:assemble"],
+            )
+            .expect_success_output("assemble SDK")
+    })?;
+
+    patch_with(PatchRuntimeWith {
+        sdk_path: args.sdk_path,
+        runtime_crate_path: smithy_rs.root.join("aws/sdk/build/aws-sdk/sdk"),
+        previous_release_tag: args.previous_release_tag,
+    })?;
+
+    Ok(())
+}
+
+pub fn patch_with(args: PatchRuntimeWith) -> Result<()> {
+    let aws_sdk_rust = step("Resolving aws-sdk-rust", || Repo::new(Some(&args.sdk_path)))?;
+    if is_dirty(&aws_sdk_rust)? {
+        bail!("aws-sdk-rust has a dirty working tree. Aborting.");
+    }
+
+    // Make sure the aws-sdk-rust repo is on the correct release tag
+    let release_tags = step("Resolving aws-sdk-rust release tags", || {
+        release_tags(&aws_sdk_rust)
+    })?;
+    let previous_release_tag = step("Resolving release tag", || {
+        previous_release_tag(
+            &aws_sdk_rust,
+            &release_tags,
+            args.previous_release_tag.as_deref(),
+        )
+    })?;
+    step("Checking out release tag", || {
+        aws_sdk_rust
+            .git(["checkout", previous_release_tag.as_str()])
+            .expect_success_output("check out release tag in aws-sdk-rust")
+    })?;
+
+    // Patch the new runtime crates into the old SDK
+    step("Applying version-only dependencies", || {
+        apply_version_only_dependencies(&aws_sdk_rust)
+    })?;
+    step("Patching aws-sdk-rust root Cargo.toml", || {
+        patch_workspace_cargo_toml(&aws_sdk_rust, &args.runtime_crate_path)
+    })?;
+    step("Running cargo update", || {
+        aws_sdk_rust
+            .cmd("cargo", ["update"])
+            .expect_success_output("cargo update")
+    })?;
+
+    Ok(())
+}
+
+fn patch_gradle_properties(smithy_rs: &Repo, args: &PatchRuntime) -> Result<()> {
+    let props_path = smithy_rs.root.join("gradle.properties");
+    let props =
+        fs::read_to_string(&props_path).context("failed to read smithy-rs/gradle.properties")?;
+    let mut new_props = String::with_capacity(props.len());
+    for line in props.lines() {
+        if line.starts_with("smithy.rs.runtime.crate.stable.version=") {
+            new_props.push_str(&format!(
+                "smithy.rs.runtime.crate.stable.version={}",
+                args.stable_crate_version
+            ));
+        } else if line.starts_with("smithy.rs.runtime.crate.unstable.version=") {
+            new_props.push_str(&format!(
+                "smithy.rs.runtime.crate.unstable.version={}",
+                args.unstable_crate_version
+            ));
+        } else {
+            new_props.push_str(line);
+        }
+        new_props.push('\n');
+    }
+    fs::write(&props_path, new_props).context("failed to write smithy-rs/gradle.properties")?;
+    Ok(())
+}
```

Comment:
can we provide this as `-P` instead? Then this will work with a dirty working tree.